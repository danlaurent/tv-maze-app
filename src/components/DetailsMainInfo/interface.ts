import { FlexAlignType } from 'react-native';

export interface IDetailsMainInfo {
  title: string;
  subtitle?: string;
  metaInfo?: string;
  rating?: number;
  align?: FlexAlignType;
}
