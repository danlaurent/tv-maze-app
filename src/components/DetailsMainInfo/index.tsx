import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { FontAwesome5 } from '@expo/vector-icons';
import { GRAY, LIGHT_GRAY } from '../../constants/styles/colors';
import { IDetailsMainInfo } from './interface';

const styles = StyleSheet.create({
  detailsName: {
    color: 'white',
    fontSize: 24,
    fontWeight: '700',
    marginBottom: 8,
    marginTop: 16,
  },
  detailsMetaInfo: {
    color: GRAY,
    marginBottom: 8,
    textAlign: 'center',
  },
  ratingContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  detailsRating: {
    fontSize: 16,
    color: LIGHT_GRAY,
    fontWeight: '700',
    marginLeft: 8,
  },
});

const DetailsMainInfo = ({
  title,
  subtitle,
  metaInfo,
  rating,
  align = 'center',
  testID,
}: IDetailsMainInfo) => (
  <View style={{ alignItems: align }}>
    <Text style={styles.detailsName}>{title}</Text>
    {subtitle ? (
      <Text testID={`${testID}_subtitle`} style={styles.detailsMetaInfo}>
        {subtitle}
      </Text>
    ) : null}
    {metaInfo ? (
      <Text testID={`${testID}_metaInfo`} style={styles.detailsMetaInfo}>
        {metaInfo}
      </Text>
    ) : null}
    <View style={styles.ratingContainer}>
      <FontAwesome5 name='star' size={16} color='yellow' />
      <Text testID={`${testID}_rating`} style={styles.detailsRating}>
        {rating || '-'}
      </Text>
    </View>
  </View>
);

export default DetailsMainInfo;
