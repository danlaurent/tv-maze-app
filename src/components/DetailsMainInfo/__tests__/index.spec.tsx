import React from 'react';
import { render, RenderAPI, fireEvent } from '@testing-library/react-native';
import DetailsMainInfo from '..';

describe('DetailsMainInfo', () => {
  let tree: RenderAPI;

  beforeEach(() => {
    tree = render(
      <DetailsMainInfo
        title='Test'
        subtitle='SubTitle Test'
        metaInfo='MetaInfo Test'
        rating={10}
        testID='testDetailsMainInfo'
      />
    );
  });
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const snapshot = tree.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('shows the title', () => {
    expect(tree.getByText('Test')).toBeDefined();
  });

  it('shows the subtitle', () => {
    expect(tree.getByText('SubTitle Test')).toBeDefined();
  });

  it('shows the metaInfo', () => {
    expect(tree.getByText('MetaInfo Test')).toBeDefined();
  });

  it('shows the rating', () => {
    expect(tree.getByText('10')).toBeDefined();
  });

  describe('When subtitle is not provided', () => {
    beforeEach(() => {
      tree = render(
        <DetailsMainInfo
          title='Test'
          metaInfo='MetaInfo Test'
          rating={10}
          testID='testDetailsMainInfo'
        />
      );
    });
    it('do not show the subtitle', () => {
      expect(tree.queryByTestId('testDetailsMainInfo_subtitle')).toBeNull();
    });
  });

  describe('When metaInfo is not provided', () => {
    beforeEach(() => {
      tree = render(
        <DetailsMainInfo
          title='Test'
          rating={10}
          testID='testDetailsMainInfo'
        />
      );
    });
    it('do not show the metaInfo', () => {
      expect(tree.queryByTestId('testDetailsMainInfo_metaInfo')).toBeNull();
    });
  });

  describe('When rating is not provided', () => {
    beforeEach(() => {
      tree = render(
        <DetailsMainInfo title='Test' testID='testDetailsMainInfo' />
      );
    });
    it('do not show the metaInfo', () => {
      expect(tree.getByTestId('testDetailsMainInfo_rating').children[0]).toBe(
        '-'
      );
    });
  });
});
