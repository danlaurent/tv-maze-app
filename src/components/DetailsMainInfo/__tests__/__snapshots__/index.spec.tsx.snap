// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsMainInfo renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 24,
        "fontWeight": "700",
        "marginBottom": 8,
        "marginTop": 16,
      }
    }
  >
    Test
  </Text>
  <Text
    style={
      Object {
        "color": "#6d6f7e",
        "marginBottom": 8,
        "textAlign": "center",
      }
    }
    testID="testDetailsMainInfo_subtitle"
  >
    SubTitle Test
  </Text>
  <Text
    style={
      Object {
        "color": "#6d6f7e",
        "marginBottom": 8,
        "textAlign": "center",
      }
    }
    testID="testDetailsMainInfo_metaInfo"
  >
    MetaInfo Test
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text />
    <Text
      style={
        Object {
          "color": "#c6c7d2",
          "fontSize": 16,
          "fontWeight": "700",
          "marginLeft": 8,
        }
      }
      testID="testDetailsMainInfo_rating"
    >
      10
    </Text>
  </View>
</View>
`;
